packages/eventual-send/src/index.d.ts

```diff
@@ -234,7 +234,7 @@ interface EProxy {
    * @param x target for method/function call
    * @returns method/function call proxy
    */
-  <T>(x: T): ECallableOrMethods<RemoteFunctions<T>>;
+  <T extends {}>(x: T): ECallableOrMethods<RemoteFunctions<T>>;
```

Comment:
Why not update `ECallableOrMethods` if extends null or undefined and return `never` (or possibly a `Record<PropertyKey, () => never>`)? That would model more closely what is actually going on, where it's the action of using the result of `E()` that is invalid, not using E itself.

Reply:
I'll look into it and see whether the feedback is better. I take it the change wouldn't affect what actually passes typecheck.

Do you have the same feedback for the `get()` change?

Reply:
I guess the only difference would be that `const p = E(Promise.resolve(null))` is entirely valid, but that `p.foo()` is the operation that is type invalid (However it is runtime valid, just results in a rejected promise (hence the `Record<PropertyKey, () => Promise<never>>` suggestion).

I think `E.get()` should be similar yes.

Reply:
The `never` is perhaps more accurate, but I wonder if constraining the input is more useful. Here's the error message with the PR version:
<img width="782" alt="Screen Shot 2022-06-23 at 2 10 53 PM" src="https://user-images.githubusercontent.com/21505/175400934-e9defa47-5ec1-484f-9852-f614b63153f2.png">

Basically it's saying, "if you're trying to use a method then the argument has to be defined". 

Using `never` gives this feedback,
<img width="443" alt="Screen Shot 2022-06-23 at 2 09 27 PM" src="https://user-images.githubusercontent.com/21505/175401092-ad0b47af-f383-4eb1-9f56-f6ae7a6ec757.png">
which doesn't give the user much clue why or how to remedy it.

The `never` is more sound than the generic constraint in that `E(undefined)` would error in TS and not in runtime (is that true?). Though are their valid cases for `E(undefined)` or even `E(Promise<undefined>)`? If those aren't runtime errors now I wonder if they should be.

Reply:
I think where it matters most if for `E(Promise<SomeRemotable | undefined>)`. For anything where the target is awaited first, the author should do a local check first. Maybe we could have the best of both worlds by doing `extends {}` and have the second check in `ECallableOrMethods` ?